### PR TITLE
HDDS-1260. Create Recon Server lifecycle integration with Ozone.

### PR DESCRIPTION
**Pull request by vivekratnavel**
- Create the lifecycle scripts (start/stop) for Recon Server along with Shell interface like the other components.
- Verify configurations are being picked up by Recon Server on startup.

### CODE REVIEW

**Comment by vivekratnavel** [comment]
@avijayanhwx @bharatviswa504 @swagle Please review when you find time

File: hadoop-ozone/ozone-recon/src/main/resources/webapps.recon.WEB-INF/web.xml [hunk 1]
```diff
@@ -1,33 +0,0 @@
-<!--
-  Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (avijayanhwx) [inline hadoop-ozone/ozone-recon/src/main/resources/webapps.recon.WEB-INF/web.xml]: Why is this file deleted?

File: hadoop-ozone/ozone-recon/pom.xml [hunk 2]
```diff
@@ -84,4 +84,33 @@
...
+            <id>copy-common-html</id>
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>unpack</goal>
+            </goals>
+            <configuration>
+              <artifactItems>
+                <artifactItem>
+                  <groupId>org.apache.hadoop</groupId>
+                  <artifactId>hadoop-hdds-server-framework</artifactId>
```
[the remark this answers is not in the document]
Comment (vivekratnavel) [reply hadoop-ozone/ozone-recon/pom.xml]: Yes. The hadoop-hdds-server-framework has commonly used static js and css files.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 59 | Maven dependency ordering for branch |
| +1 | mvninstall | 996 | trunk passed |
| +1 | compile | 107 | trunk passed |
| +1 | checkstyle | 35 | trunk passed |
| +1 | mvnsite | 150 | trunk passed |
| +1 | shadedclient | 667 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 141 | trunk passed |
| +1 | javadoc | 119 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| -1 | mvninstall | 21 | dist in the patch failed. |
| +1 | compile | 96 | the patch passed |
| +1 | javac | 96 | the patch passed |
| +1 | checkstyle | 25 | the patch passed |
| +1 | mvnsite | 112 | the patch passed |
| +1 | shellcheck | 25 | There were no new shellcheck issues. |
| +1 | shelldocs | 18 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 762 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 160 | the patch passed |
| +1 | javadoc | 104 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 36 | common in the patch passed. |
| +1 | unit | 45 | ozone-manager in the patch passed. |
| +1 | unit | 36 | ozone-recon in the patch passed. |
| +1 | unit | 25 | dist in the patch passed. |
| +1 | asflicense | 33 | The patch does not generate ASF License warnings. |
| | | 4087 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 703a0a4ae7c5 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c99b107 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/2/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/2/testReport/ |
| Max. process+thread count | 441 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 28 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 63 | Maven dependency ordering for branch |
| +1 | mvninstall | 1031 | trunk passed |
| +1 | compile | 121 | trunk passed |
| +1 | checkstyle | 27 | trunk passed |
| +1 | mvnsite | 132 | trunk passed |
| +1 | shadedclient | 639 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 134 | trunk passed |
| +1 | javadoc | 95 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for patch |
| -1 | mvninstall | 18 | dist in the patch failed. |
| +1 | compile | 92 | the patch passed |
| +1 | javac | 92 | the patch passed |
| +1 | checkstyle | 23 | the patch passed |
| +1 | mvnsite | 98 | the patch passed |
| +1 | shellcheck | 24 | There were no new shellcheck issues. |
| +1 | shelldocs | 15 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 719 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 155 | the patch passed |
| +1 | javadoc | 91 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 36 | common in the patch passed. |
| -1 | unit | 43 | ozone-manager in the patch failed. |
| +1 | unit | 34 | ozone-recon in the patch passed. |
| +1 | unit | 21 | dist in the patch passed. |
| +1 | asflicense | 28 | The patch does not generate ASF License warnings. |
| | | 3921 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.ozone.security.TestOzoneDelegationTokenSecretManager |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 65b968f0f31e 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c99b107 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/3/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
| unit | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/3/artifact/out/patch-unit-hadoop-ozone_ozone-manager.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/3/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 511 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 56 | Maven dependency ordering for branch |
| +1 | mvninstall | 1106 | trunk passed |
| +1 | compile | 111 | trunk passed |
| +1 | checkstyle | 29 | trunk passed |
| +1 | mvnsite | 136 | trunk passed |
| +1 | shadedclient | 623 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 151 | trunk passed |
| +1 | javadoc | 103 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for patch |
| -1 | mvninstall | 21 | dist in the patch failed. |
| +1 | compile | 101 | the patch passed |
| +1 | javac | 101 | the patch passed |
| +1 | checkstyle | 26 | the patch passed |
| +1 | mvnsite | 106 | the patch passed |
| +1 | shellcheck | 27 | There were no new shellcheck issues. |
| +1 | shelldocs | 12 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 742 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 162 | the patch passed |
| +1 | javadoc | 86 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 36 | common in the patch passed. |
| +1 | unit | 44 | ozone-manager in the patch passed. |
| +1 | unit | 34 | ozone-recon in the patch passed. |
| +1 | unit | 21 | dist in the patch passed. |
| +1 | asflicense | 31 | The patch does not generate ASF License warnings. |
| | | 4539 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 00d8f7473f37 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c99b107 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/1/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/1/testReport/ |
| Max. process+thread count | 447 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 25 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 31 | Maven dependency ordering for branch |
| +1 | mvninstall | 1066 | trunk passed |
| +1 | compile | 104 | trunk passed |
| +1 | checkstyle | 29 | trunk passed |
| -1 | mvnsite | 28 | ozone-manager in trunk failed. |
| +1 | shadedclient | 648 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| -1 | findbugs | 27 | ozone-manager in trunk failed. |
| +1 | javadoc | 93 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 12 | Maven dependency ordering for patch |
| -1 | mvninstall | 19 | dist in the patch failed. |
| +1 | compile | 99 | the patch passed |
| +1 | javac | 99 | the patch passed |
| +1 | checkstyle | 24 | the patch passed |
| +1 | mvnsite | 102 | the patch passed |
| +1 | shellcheck | 25 | There were no new shellcheck issues. |
| +1 | shelldocs | 13 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 5 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 739 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 156 | the patch passed |
| +1 | javadoc | 86 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 35 | common in the patch passed. |
| +1 | unit | 40 | ozone-manager in the patch passed. |
| +1 | unit | 37 | ozone-recon in the patch passed. |
| +1 | unit | 21 | dist in the patch passed. |
| +1 | asflicense | 27 | The patch does not generate ASF License warnings. |
| | | 3921 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 1a4c3caa7d66 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c99b107 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| mvnsite | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/4/artifact/out/branch-mvnsite-hadoop-ozone_ozone-manager.txt |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| findbugs | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/4/artifact/out/branch-findbugs-hadoop-ozone_ozone-manager.txt |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/4/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/4/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/4/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 65 | Maven dependency ordering for branch |
| +1 | mvninstall | 1065 | trunk passed |
| +1 | compile | 109 | trunk passed |
| +1 | checkstyle | 31 | trunk passed |
| +1 | mvnsite | 133 | trunk passed |
| +1 | shadedclient | 633 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 146 | trunk passed |
| +1 | javadoc | 107 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| -1 | mvninstall | 23 | dist in the patch failed. |
| +1 | compile | 111 | the patch passed |
| +1 | javac | 111 | the patch passed |
| +1 | checkstyle | 25 | the patch passed |
| +1 | mvnsite | 111 | the patch passed |
| +1 | shellcheck | 27 | There were no new shellcheck issues. |
| +1 | shelldocs | 13 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 3 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 728 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 162 | the patch passed |
| +1 | javadoc | 108 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 39 | common in the patch passed. |
| +1 | unit | 45 | ozone-manager in the patch passed. |
| +1 | unit | 37 | ozone-recon in the patch passed. |
| +1 | unit | 26 | dist in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 4090 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 048df944be83 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c99b107 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/5/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/5/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/5/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 511 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 62 | Maven dependency ordering for branch |
| +1 | mvninstall | 1081 | trunk passed |
| +1 | compile | 103 | trunk passed |
| +1 | checkstyle | 26 | trunk passed |
| +1 | mvnsite | 130 | trunk passed |
| -1 | shadedclient | 1044 | branch has errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 145 | trunk passed |
| +1 | javadoc | 105 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 16 | Maven dependency ordering for patch |
| -1 | mvninstall | 22 | dist in the patch failed. |
| +1 | compile | 96 | the patch passed |
| +1 | javac | 96 | the patch passed |
| +1 | checkstyle | 24 | the patch passed |
| +1 | mvnsite | 107 | the patch passed |
| +1 | shellcheck | 24 | There were no new shellcheck issues. |
| +1 | shelldocs | 15 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 4 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 744 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 159 | the patch passed |
| +1 | javadoc | 105 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 38 | common in the patch passed. |
| +1 | unit | 46 | ozone-manager in the patch passed. |
| +1 | unit | 35 | ozone-recon in the patch passed. |
| +1 | unit | 24 | dist in the patch passed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 4961 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux c580f94eed50 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c99b107 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/6/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/6/testReport/ |
| Max. process+thread count | 411 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-manager hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/6/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by avijayanhwx** [comment]
LGTM +1. We can build a proper UI to serve up container-key mapping in [HDDS-1335](https://issues.apache.org/jira/browse/HDDS-1335)

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 7 | https://github.com/apache/hadoop/pull/643 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/7/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 47 | Maven dependency ordering for branch |
| +1 | mvninstall | 970 | trunk passed |
| +1 | compile | 103 | trunk passed |
| +1 | checkstyle | 26 | trunk passed |
| +1 | mvnsite | 95 | trunk passed |
| +1 | shadedclient | 625 | branch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 93 | trunk passed |
| +1 | javadoc | 77 | trunk passed |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 15 | Maven dependency ordering for patch |
| -1 | mvninstall | 18 | dist in the patch failed. |
| +1 | compile | 90 | the patch passed |
| +1 | javac | 90 | the patch passed |
| +1 | checkstyle | 22 | the patch passed |
| +1 | mvnsite | 72 | the patch passed |
| +1 | shellcheck | 24 | There were no new shellcheck issues. |
| +1 | shelldocs | 11 | There were no new shelldocs issues. |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | xml | 2 | The patch has no ill-formed XML file. |
| +1 | shadedclient | 695 | patch has no errors when building and testing our client artifacts. |
| 0 | findbugs | 0 | Skipped patched modules with no Java source: hadoop-ozone/dist |
| +1 | findbugs | 106 | the patch passed |
| +1 | javadoc | 68 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 34 | common in the patch passed. |
| +1 | unit | 32 | ozone-recon in the patch passed. |
| +1 | unit | 20 | dist in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 3521 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-643/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/643 |
| Optional Tests |  dupname  asflicense  mvnsite  unit  shellcheck  shelldocs  compile  javac  javadoc  mvninstall  shadedclient  xml  findbugs  checkstyle  |
| uname | Linux 69eb151f6f53 4.4.0-139-generic #165-Ubuntu SMP Wed Oct 24 10:58:50 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 56f1e13 |
| maven | version: Apache Maven 3.3.9 |
| Default Java | 1.8.0_191 |
| shellcheck | v0.4.6 |
| findbugs | v3.1.0-RC1 |
| mvninstall | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/8/artifact/out/patch-mvninstall-hadoop-ozone_dist.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/8/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone/common hadoop-ozone/ozone-recon hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-643/8/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by vivekratnavel** [comment]
Robot tests will be added later as part of - https://issues.apache.org/jira/browse/HDDS-1261
Tested build manually and it succeeds.
